src/App.tsx

```diff
@@ -21,38 +34,53 @@ export const App: React.FC = () => {
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">No user selected</p>
-
-                <Loader />
-
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
-
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
+                {!selectedUser && (
+                  <p data-cy="NoSelectedUser">No user selected</p>
+                )}
 
-                <PostsList />
+                {selectedUser && (
+                  <>
+                    {isPostsLoading ? (
+                      <Loader />
+                    ) : (
+                      <>
+                        {postsError ? (
+                          <div
+                            className="notification is-danger"
+                            data-cy="PostsLoadingError"
+                          >
+                            Something went wrong!
+                          </div>
+                        ) : (
+                          <>
+                            {!posts.length ? (
+                              <div
+                                className="notification is-warning"
+                                data-cy="NoPostsYet"
+                              >
+                                No posts yet
+                              </div>
+                            ) : (
+                              <PostsList />
+                            )}
+                          </>
+                        )}
+                      </>
+                    )}
+                  </>
+                )}
```

Comment:
looks very complicated, let's try to simplify it using `&&` operator, for ex:
`!isPostsLoading && !!selectedUser && hasPosts && <PostsList />`